### ISSUE TITLE
Interpreter mode selection

### ISSUE THREAD

**Issue by vshymanskyy** [wasm3/wasm3#241]
On some platforms, `TCO` (tail call elimination) is not available/implemented, so we need an alternative VM execution mode.
Some work was done #74, however this won't be merged (POC, needs too much work to complete).
Creating this ticket to track progress of this feature.

**Comment by sascha1337** [comment]
YES progress! do i understand correctly this could possibly help to get wasi to non SPRAM ESP32's?

**Comment by sfranzyshen** [comment]
After reading the following statement in this [document](https://github.com/wasm3/wasm3/blob/main/docs/Interpreter.md#stack-usage) from the wasm3 project ... I decided to try something different with my failing code ... and now it's working on both the ESP32 and esp8266 platforms ... No more out of stack 

```Returning from a Wasm function also unwinds the stack, back to the point of the Call operation.```

the work around for wasm3 issue [#241](https://github.com/wasm3/wasm3/issues/241) is to write the code inside the wasm app to be non blocking ... allowing the execution flow to return back to the arduino host, resetting the stack used ... **bingo!** 

here's an example working on both esp8266 & esp32 ... https://github.com/sfranzyshen/wasm3_testing

**Comment by vshymanskyy** [comment]
I have to admit that all prototypes and development of this feature were lost when my house was destroyed by Russians.
So unfortunately I won't be able to continue working on this, at least for some time. Should have made more backups, sorry.
